<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" SYSTEM "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"><head><meta content="text/html; charset=utf-8" http-equiv="Content-Type"/><title>numpy.errstate — NumPy v1.7.dev-3f685a1 Manual (DRAFT)</title><link href="../../_static/scipy.css" rel="stylesheet" type="text/css"/><link href="../../_static/pygments.css" rel="stylesheet" type="text/css"/><script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../',
        VERSION:     '1.7.0.dev-3f685a1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script><script src="../../_static/jquery.js" type="text/javascript"></script><script src="../../_static/underscore.js" type="text/javascript"></script><script src="../../_static/doctools.js" type="text/javascript"></script><link href="../../about.html" rel="author" title="About these documents"/><link href="../../index.html" rel="top" title="NumPy v1.7.dev-3f685a1 Manual (DRAFT)"/><link href="../routines.err.html" rel="up" title="Floating point error handling"/><link href="numpy.seterrobj.html" rel="next" title="numpy.seterrobj"/><link href="numpy.geterrcall.html" rel="prev" title="numpy.geterrcall"/></head><body>
<div class="related">
<h3>Navigation</h3>
<ul>
<li class="right" style="margin-right: 10px">
<a accesskey="I" href="../../genindex.html" title="General Index">index</a></li>
<li class="right">
<a href="../../py-modindex.html" title="Python Module Index">modules</a> |</li>
<li class="right">
<a href="../../np-modindex.html" title="Python Module Index">modules</a> |</li>
<li class="right">
<a accesskey="N" href="numpy.seterrobj.html" title="numpy.seterrobj">next</a> |</li>
<li class="right">
<a accesskey="P" href="numpy.geterrcall.html" title="numpy.geterrcall">previous</a> |</li>
<li><a href="../../index.html">NumPy v1.7.dev-3f685a1 Manual (DRAFT)</a> »</li>
<li><a href="../index.html">NumPy Reference</a> »</li>
<li><a href="../routines.html">Routines</a> »</li>
<li><a accesskey="U" href="../routines.err.html">Floating point error handling</a> »</li>
</ul>
</div>
<div class="document">
<div class="documentwrapper">
<div class="bodywrapper">
<div class="body">
<div class="section" id="numpy-errstate">
<h1>numpy.errstate<a class="headerlink" href="#numpy-errstate" title="Permalink to this headline">¶</a></h1>
<dl class="class">
<a name="//apple_ref/cpp/cl/numpy.errstate"></a><dt id="numpy.errstate">
<em class="property">class </em><tt class="descclassname">numpy.</tt><tt class="descname">errstate</tt><big>(</big><em>**kwargs</em><big>)</big><a class="headerlink" href="#numpy.errstate" title="Permalink to this definition">¶</a></dt>
<dd><p>Context manager for floating-point error handling.</p>
<p>Using an instance of <a class="reference internal" href="#numpy.errstate" title="numpy.errstate"><tt class="xref py py-obj docutils literal"><span class="pre">errstate</span></tt></a> as a context manager allows statements in
that context to execute with a known error handling behavior. Upon entering
the context the error handling is set with <a class="reference internal" href="numpy.seterr.html#numpy.seterr" title="numpy.seterr"><tt class="xref py py-obj docutils literal"><span class="pre">seterr</span></tt></a> and <a class="reference internal" href="numpy.seterrcall.html#numpy.seterrcall" title="numpy.seterrcall"><tt class="xref py py-obj docutils literal"><span class="pre">seterrcall</span></tt></a>, and
upon exiting it is reset to what it was before.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name"></col>
<col class="field-body"></col>
<tbody valign="top">
<tr class="field"><th class="field-name">Parameters :</th><td class="field-body"><p class="first"><strong>kwargs</strong> : {divide, over, under, invalid}</p>
<blockquote class="last">
<div><p>Keyword arguments. The valid keywords are the possible floating-point
exceptions. Each keyword should have a string value that defines the
treatment for the particular error. Possible values are
{‘ignore’, ‘warn’, ‘raise’, ‘call’, ‘print’, ‘log’}.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
<div class="admonition-see-also admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="numpy.seterr.html#numpy.seterr" title="numpy.seterr"><tt class="xref py py-obj docutils literal"><span class="pre">seterr</span></tt></a>, <a class="reference internal" href="numpy.geterr.html#numpy.geterr" title="numpy.geterr"><tt class="xref py py-obj docutils literal"><span class="pre">geterr</span></tt></a>, <a class="reference internal" href="numpy.seterrcall.html#numpy.seterrcall" title="numpy.seterrcall"><tt class="xref py py-obj docutils literal"><span class="pre">seterrcall</span></tt></a>, <a class="reference internal" href="numpy.geterrcall.html#numpy.geterrcall" title="numpy.geterrcall"><tt class="xref py py-obj docutils literal"><span class="pre">geterrcall</span></tt></a></p>
</div>
<p class="rubric">Notes</p>
<p>The <tt class="docutils literal"><span class="pre">with</span></tt> statement was introduced in Python 2.5, and can only be used
there by importing it: <tt class="docutils literal"><span class="pre">from</span> <span class="pre">__future__</span> <span class="pre">import</span> <span class="pre">with_statement</span></tt>. In
earlier Python versions the <tt class="docutils literal"><span class="pre">with</span></tt> statement is not available.</p>
<p>For complete documentation of the types of floating-point exceptions and
treatment options, see <a class="reference internal" href="numpy.seterr.html#numpy.seterr" title="numpy.seterr"><tt class="xref py py-obj docutils literal"><span class="pre">seterr</span></tt></a>.</p>
<p class="rubric">Examples</p>
<div class="highlight-python"><div class="highlight"><pre><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">with_statement</span>  <span class="c"># use 'with' in Python 2.5</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">olderr</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">seterr</span><span class="p">(</span><span class="nb">all</span><span class="o">=</span><span class="s">'ignore'</span><span class="p">)</span>  <span class="c"># Set error handling to known state.</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="gp">&gt;&gt;&gt; </span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span> <span class="o">/</span> <span class="mf">0.</span>
<span class="go">array([ NaN,  Inf,  Inf])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="n">np</span><span class="o">.</span><span class="n">errstate</span><span class="p">(</span><span class="n">divide</span><span class="o">=</span><span class="s">'warn'</span><span class="p">):</span>
<span class="gp">... </span>    <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span> <span class="o">/</span> <span class="mf">0.</span>
<span class="gp">...</span>
<span class="go">__main__:2: RuntimeWarning: divide by zero encountered in divide</span>
<span class="go">array([ NaN,  Inf,  Inf])</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="gp">&gt;&gt;&gt; </span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="o">-</span><span class="mi">1</span><span class="p">)</span>
<span class="go">nan</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="n">np</span><span class="o">.</span><span class="n">errstate</span><span class="p">(</span><span class="n">invalid</span><span class="o">=</span><span class="s">'raise'</span><span class="p">):</span>
<span class="gp">... </span>    <span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="o">-</span><span class="mi">1</span><span class="p">)</span>
<span class="gt">Traceback (most recent call last):</span>
  File <span class="nb">"&lt;stdin&gt;"</span>, line <span class="m">2</span>, in <span class="n">&lt;module&gt;</span>
<span class="gr">FloatingPointError</span>: <span class="n">invalid value encountered in sqrt</span>
</pre></div>
</div>
<p>Outside the context the error handling behavior has not changed:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="gp">&gt;&gt;&gt; </span><span class="n">np</span><span class="o">.</span><span class="n">geterr</span><span class="p">()</span>
<span class="go">{'over': 'warn', 'divide': 'warn', 'invalid': 'warn',</span>
<span class="go">'under': 'ignore'}</span>
</pre></div>
</div>
</dd></dl>
</div>
</div>
</div>
</div>
<div class="sphinxsidebar">
<div class="sphinxsidebarwrapper">
<p class="logo"><a href="../../contents.html">
<img alt="Logo" class="logo" src="../../_static/scipyshiny_small.png"/>
</a></p>
<h4>Previous topic</h4>
<p class="topless"><a href="numpy.geterrcall.html" title="previous chapter">numpy.geterrcall</a></p>
<h4>Next topic</h4>
<p class="topless"><a href="numpy.seterrobj.html" title="next chapter">numpy.seterrobj</a></p>
<h3>This Page</h3>
<ul class="this-page-menu">
<li><a href="../../_sources/reference/generated/numpy.errstate.txt" rel="nofollow">Show Source</a></li>
</ul>
<ul class="this-page-menu">
<li><a href="/numpy/docs/numpy.errstate">Edit page</a></li>
</ul>
<div id="searchbox" style="display: none">
<h3>Quick search</h3>
<form action="../../search.html" class="search" method="get">
<input name="q" size="18" type="text"/>
<input type="submit" value="Go"/>
<input name="check_keywords" type="hidden" value="yes"/>
<input name="area" type="hidden" value="default"/>
</form>
<p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
</div>
</div>
<div class="clearer"></div>
</div>
<div class="related">
<h3>Navigation</h3>
<ul>
<li class="right" style="margin-right: 10px">
<a href="../../genindex.html" title="General Index">index</a></li>
<li class="right">
<a href="../../py-modindex.html" title="Python Module Index">modules</a> |</li>
<li class="right">
<a href="../../np-modindex.html" title="Python Module Index">modules</a> |</li>
<li class="right">
<a href="numpy.seterrobj.html" title="numpy.seterrobj">next</a> |</li>
<li class="right">
<a href="numpy.geterrcall.html" title="numpy.geterrcall">previous</a> |</li>
<li><a href="../../index.html">NumPy v1.7.dev-3f685a1 Manual (DRAFT)</a> »</li>
<li><a href="../index.html">NumPy Reference</a> »</li>
<li><a href="../routines.html">Routines</a> »</li>
<li><a href="../routines.err.html">Floating point error handling</a> »</li>
</ul>
</div>
<div class="footer">
        © Copyright 2008-2009, The Scipy community.
      Last updated on Feb 24, 2012.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.0.7.
    </div>
</body></html>